activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -930,7 +930,19 @@ def set_value(name, value) # :nodoc:
 
       # Extract column names from arguments passed to #order or #reorder.
       def column_names_from_order_arguments(args)
-        args.flat_map { |arg| arg.is_a?(Hash) ? arg.keys : arg }
+        args.flat_map do |arg|
+          case arg
+          when Hash
+            # Tag.order(id: :desc)
+            arg.keys
+          when Array
+            # Tag.order([Arel.sql("field(id, ?)"), [1, 3, 2]])
+            arg.flatten
```

Comment:
`arg.first`?

Reply:
Nope. Unfortunately, people can also pass `Tag.order(["id", "sql", "injection"])`

Reply:
Sometimes I'm amazed by the flexibility of Rails' APIs:

```
> Tag.order([[[[["id"], "name"], "taggings_count"]]]).first
D, [2017-10-13T08:13:06.869574 #58904] DEBUG -- :   Tag Load (0.2ms)  SELECT  "tags".* FROM "tags" ORDER BY id, name, taggings_count LIMIT ?  [["LIMIT", 1]]
=> #<Tag:0x00007ff310a3dbc0 id: 1, name: "General", taggings_count: 0>
```

Reply:
Oh, wow! That complicates things a bit... meanwhile `Tag.order([Arel.sql("length(?)"), "something (suspicious) here"])` should work without complaint, because the alternative would be to demand a lie [that that string was SQL].

Perhaps we'd be better off pushing the check a bit further down the `preprocess_order_args` / `sanitize_sql_for_order` path, so we can leverage the existing destructuring/parsing, instead of having to mirror it?